### PR TITLE
More fixes for Azure hosting

### PR DESCRIPTION
**Pull request by sweetlilmre**
### Background
This pull request updates Azure compatibility for those wishing to run AutoGPT against an Azure deployment.

### Changes
- Updates to allow the embeddings code to work with Azure, specifically to point to a separately deployed model for `text-embedding-ada-002` 
- moves azure configuration into a separate file to stop the pollution of .env

### Documentation
Changes for the user (additional .env variable required, azure.yaml configuration and link to an Azure page giving detail) are supplied in the README.md

### Test Plan
Ran this against Azure with a GPT-4 and text-embedding-ada-002 deployment on 2 separate deployment ids, with the stock setup and other prompts.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thouroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes 

Look, I'm no expert here, I've tested this against a few prompts and it seems to work fine. I'm hoping I'm not the only person running this against an Azure deployment and that others more knowledgeable than me can chime in and check it out. The code path should be clean enough that it doesn't cause any side effects to the main code base.

### CODE REVIEW

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — APPROVED** [review]

**Comment by sweetlilmre** [comment]
@nponeccop I've made some changes which clean up the azure implementation and effectively combine and replace [#700](https://github.com/Torantulino/Auto-GPT/pull/700) and #580

File: .env.template [hunk 1]
```diff
@@ -7,9 +7,6 @@ FAST_LLM_MODEL=gpt-3.5-turbo
 GOOGLE_API_KEY=
 CUSTOM_SEARCH_ENGINE_ID=
 USE_AZURE=False
-OPENAI_AZURE_API_BASE=your-base-url-for-azure
-OPENAI_AZURE_API_VERSION=api-version-for-azure
-OPENAI_AZURE_DEPLOYMENT_ID=deployment-id-for-azure
```
Comment (nponeccop) [inline .env.template:16]: If this is intended I'm ok

Comment (sweetlilmre) [reply .env.template:16]: 100% intended :) configuration for azure has move to azure.yaml to stop pollution of the .env file

**Review by nponeccop — APPROVED** [review]

**Review by suensummit — APPROVED** [review]
nicely done 👍

**Comment by nponeccop** [comment]
@sweetlilmre There are conflicts now

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@sweetlilmre Now there are no conflicts, but you need to fix CI errors

**Comment by nponeccop** [comment]
@sweetlilmre There are conflicts and failing CI checks now

**Comment by sweetlilmre** [comment]
I'm getting hit by a merge storm :)

**Comment by sweetlilmre** [comment]
@[richbeales](https://github.com/richbeales) When you get a chance, this supersedes #700 and #580 and cleans up azure config

**Review by nponeccop — APPROVED** [review]

**Review by richbeales — APPROVED** [review]

**Pull request merged**